src/k8s-engine/client.js

```diff
@@ -360,6 +360,21 @@ class KubeClient extends events.EventEmitter {
     return this.#servers.get(namespace, endpoint, port)?.address()?.port;
   }
 
+  /**
+   * Return a list of all the pods with the given app-label in the specified namespace
+   * @param namespace
+   * @param endpointName
+   * @returns {Promise<*[{{ name: string, status: string }}]>}
+   */
+  async listPods(namespace, endpointName) {
+    console.log(`Attempting to locate ${ namespace }/${ endpointName } labelSelector: app=${ endpointName } pod...`);
+    const result = await this.#coreV1API.listNamespacedPod(namespace, undefined, undefined, undefined, undefined, `app=${ endpointName }`);
```

Comment:
The label key `app` is not a standard by any means. Some people use it some of the time. The current documented labels are https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Would it make sense for this function to take a list of labels and the actual label used it pushed to the rancher/homestead handling code?

Reply:
I'm looking for at least one pod with the name `rancher-someRandomSuffix` -- the homestead helm chart specifies a label `app: {{ template "rancher.fullname" . }}` for the deployment, so this looked right to me.

Should the helm chart instead use `app.kubernetes.io/name: {{ template "rancher.fullname" . }}`

The part I don't understand about your question is that this code isn't generic code -- it's looking for something we have full control over.